src/source/Ice/IceAgent.c

```diff
@@ -1166,19 +1189,23 @@ STATUS createIceCandidatePairs(PIceAgent pIceAgent, PIceCandidate pIceCandidate,
             CHK_STATUS(hashTableCreateWithParams(ICE_HASH_TABLE_BUCKET_COUNT, ICE_HASH_TABLE_BUCKET_LENGTH, &pIceCandidatePair->requestSentTime));
 
             pIceCandidatePair->lastDataSentTime = 0;
-            STRNCPY(pIceCandidatePair->rtcIceCandidatePairDiagnostics.localCandidateId, pIceCandidatePair->local->id,
-                    ARRAY_SIZE(pIceCandidatePair->rtcIceCandidatePairDiagnostics.localCandidateId));
-            STRNCPY(pIceCandidatePair->rtcIceCandidatePairDiagnostics.remoteCandidateId, pIceCandidatePair->remote->id,
-                    ARRAY_SIZE(pIceCandidatePair->rtcIceCandidatePairDiagnostics.remoteCandidateId));
-            pIceCandidatePair->rtcIceCandidatePairDiagnostics.state = pIceCandidatePair->state;
-            pIceCandidatePair->rtcIceCandidatePairDiagnostics.nominated = pIceCandidatePair->nominated;
-            pIceCandidatePair->rtcIceCandidatePairDiagnostics.lastPacketSentTimestamp = pIceCandidatePair->lastDataSentTime;
             pIceCandidatePair->firstStunRequest = TRUE;
             pIceCandidatePair->priority = computeCandidatePairPriority(pIceCandidatePair, pIceAgent->isControlling);
-            pIceCandidatePair->rtcIceCandidatePairDiagnostics.totalRoundTripTime = 0.0;
-            pIceCandidatePair->rtcIceCandidatePairDiagnostics.currentRoundTripTime = 0.0;
-            // Set data sending ICE candidate pair stats
-            NULLABLE_SET_EMPTY(pIceCandidatePair->rtcIceCandidatePairDiagnostics.circuitBreakerTriggerCount);
+
+            if (pIceCandidatePair->pRtcIceCandidatePairDiagnostics != NULL) {
```

Comment:
If the flag isnt enabled, the allocation does not happen which means these objects would be NULL. So it means the same thing. NULL is a more full proof check since it also covers situations where the object is deallocated in free. We do not need both checks.

Reply:
Got that, but want to point out that we are not doing a STATUS_NOT_ENOUGH_MEMORY check when allocating for `RtcIceCandidatePairDiagnostics` as we do for the allocation of other diagnostics-related structs.

Reply:
Good catch. Done.